internal/testutils/podbuilder/podbuilder.go

```diff
@@ -371,6 +387,28 @@ func (b PodBuilder) validateContainerIDs(numContainers int) {
 	}
 }
 
+func (b PodBuilder) determineNamespace(entities []k8s.K8sEntity) k8s.Namespace {
+	// N.B. we want to be careful to not coerce values to `default`, which might not be the implicit default based
+	// 	on configured context, so we really want to leave this as an empty string in that case
+	nsVal := entities[0].NamespaceOrDefault("")
+	for i := 1; i < len(entities)-1; i++ {
+		if string(entities[i].Namespace()) != nsVal {
+			b.t.Fatalf("PodBuilder only works with Manifests that reference exactly 1 namespace (found %s and %s)",
+				nsVal, entities[i].Namespace())
+		}
+	}
+	if nsVal != "" {
+		return k8s.Namespace(nsVal)
+	}
+	if b.contextNamespace != "" {
+		return b.contextNamespace
+	}
+	// this is actually redundant as long as k8s.Namespace::String() is used which
+	// coerces empty string to this, but that behavior is actually a bit sketchy,
```

Comment:
:nods:

Reply:
would really like to get rid of k8s.Namespace, in retrospect it was a mistake and doesn't play well with other k8s code we have to interop with

Reply:
I'd seen your (very helpful) comment on it previously which is what led me to adding this 😅